app/models/concerns/profile_policy_association.rb

```diff
@@ -52,7 +52,7 @@ def destroy_policy_with_internal
     end
 
     def destroy_empty_policy
-      return unless policy&.profiles&.empty?
+      return if !(policy&.profiles&.empty? && policy&.profiles&.all? { |p| p.destroyed? })
```

Comment:
`any?` and `none?` maybe?

Reply:
@skateman
To clarify, is the idea to proceed if there is _at least one_ destroyed profile?
Shouldn't we only proceed if _all_ profiles are destroyed?

Reply:
We're not changing any feature in this PR, my hint was only to reduce the complexity of the conditions you created.